### PR TITLE
[AKS] `az aks create/update`: Add `--network-policy none` option to command

### PR DESCRIPTION
**Pull request by robogatikov**
**Related command**
az aks create/update

**Description**
We are adding --network-policy=none option. "none" is currently the default value when network policy is not specified. This change is needed to support Uninstall Network Policy Engine feature (Azure Network Policy Manager or Calico)

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/_help.py [hunk 1]
```diff
@@ -256,11 +256,14 @@
...
     type: string
-    short-summary: The Kubernetes network policy to use.
+    short-summary: Network Policy engine to use.
     long-summary: |
-        Using together with "azure" network plugin.
-        Specify "azure" for Azure network policy manager, "calico" for calico network policy controller, "cilium" for Azure CNI powered by Cilium.
-        Defaults to "" (network policy disabled).
+        Azure provides three Network Policy engines for enforcing network policies that can be used together with "azure" network plugin. The following values can be specified:
+          - "azure" for Azure Network Policy Manager,
+          - "cilium" for Azure CNI Overlay powered by Cilium,
```
Comment (wedaly) [inline src/azure-cli/azure/cli/command_modules/acs/_help.py:263]: Product name is "Azure CNI Powered by Cilium" (works with both overlay and podsubnet)

File: src/azure-cli/azure/cli/command_modules/acs/_help.py [hunk 2]
```diff
@@ -256,11 +256,14 @@
...
+          - "calico" for open-source network and network security solution founded by Tigera,
+          - "none" to uninstall Azure Network Policy Manager or Calico.
```
Comment (wedaly) [inline src/azure-cli/azure/cli/command_modules/acs/_help.py:265]: "none" means no network policy engine installed, not necessarily uninstall

**Review by wedaly — CHANGES_REQUESTED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/_help.py (hunk 1 above)
Comment (robogatikov) [reply src/azure-cli/azure/cli/command_modules/acs/_help.py]: Fixed to "Azure CNI Powered by Cilium"

File: src/azure-cli/azure/cli/command_modules/acs/_help.py (hunk 2 above)
Comment (robogatikov) [reply src/azure-cli/azure/cli/command_modules/acs/_help.py]: Right, thanks for the catch. Updated descriptions for "none" network policy in both create and update routes.

File: src/azure-cli/azure/cli/command_modules/acs/_help.py (hunk 1 above)
Comment (wedaly) [reply src/azure-cli/azure/cli/command_modules/acs/_help.py]: still says Overlay?

Comment (robogatikov) [reply src/azure-cli/azure/cli/command_modules/acs/_help.py]: Sorry, fixed

**Review by wedaly — APPROVED** [review]

**Comment by yonzhan** [comment]
Please fix CI issues

File: src/azure-cli/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,12 @@
 Release History
 ===============
 
+2.63.0
+++++++
+
+**AKS**
+* `az aks create/update`: Add `--network-policy=none` support.
+
```
Comment (FumingZhang) [inline src/azure-cli/HISTORY.rst:11]: Don't need to manually edit the release history in official azure-cli. They'll generate the note from PR title or the History Notes section in PR description. So please update your PR title.

**Review by FumingZhang — COMMENTED** [review]
lgtm

Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=98702459&view=results).

- test_aks_uninstall_azure_npm
- test_aks_install_azure_npm
- test_aks_uninstall_calico_npm
- test_aks_install_calico_npm

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 4]
```diff
@@ -8479,6 +8479,246 @@ def test_aks_migrate_cluster_to_cilium_dataplane(self, resource_group, resource_
         self.cmd(
             'aks delete -g {resource_group} -n {name} --yes --no-wait', checks=[self.is_empty()])
 
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(
+        random_name_length=17,
+        name_prefix="clitest",
+        location="eastus",
+        preserve_default_location=True,
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:8487]: Please remove this parameter `preserve_default_location=True,` if this case could be executed in other location.

The test sub could not create vm in this region right now.

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 5]
```diff
@@ -8479,6 +8479,246 @@ def test_aks_migrate_cluster_to_cilium_dataplane(self, resource_group, resource_
...
+            "aks delete -g {resource_group} -n {name} --yes --no-wait",
+            checks=[self.is_empty()],
+        )
+
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(
+        random_name_length=17,
+        name_prefix="clitest",
+        location="eastus",
+        preserve_default_location=True,
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:8547]: same here

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 6]
```diff
@@ -8479,6 +8479,246 @@ def test_aks_migrate_cluster_to_cilium_dataplane(self, resource_group, resource_
...
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:8607]: same here

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 7]
```diff
@@ -8479,6 +8479,246 @@ def test_aks_migrate_cluster_to_cilium_dataplane(self, resource_group, resource_
...
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:8667]: same here

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py (hunk 4 above)
Comment (robogatikov) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py]: Thank you, @FumingZhang , removed parameter `preserve_default_location=True`.

File: src/azure-cli/HISTORY.rst (hunk 3 above)
Comment (robogatikov) [reply src/azure-cli/HISTORY.rst]: Thanks, removed changes to HISTORY.rst

**Comment by FumingZhang** [comment]
> lgtm, please rerun CI checks (e.g., push an empty commit).
> 
> Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=98711564&view=results).
> 
> * test_aks_uninstall_azure_npm
> * test_aks_install_azure_npm
> * test_aks_uninstall_calico_npm
> * test_aks_install_calico_npm

The live test failed due to the feature requires API version to be >= 2024-05-01. Need to wait until #29436 is merged.

**Comment by FumingZhang** [comment]
> > lgtm, please rerun CI checks (e.g., push an empty commit).
> > Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=98711564&view=results).
> > 
> > * test_aks_uninstall_azure_npm
> > * test_aks_install_azure_npm
> > * test_aks_uninstall_calico_npm
> > * test_aks_install_calico_npm
> 
> The live test failed due to the feature requires API version to be >= 2024-05-01. Need to wait until #29436 is merged.

#29436 is merged. Please rebase from main to pick up the change and update the recording file in your PR accordingly. @robogatikov

**Review by wedaly — APPROVED** [review]

**Comment by FumingZhang** [comment]
Seems the recording files you uploaded were generated with aks-preview installed : )

> E           AssertionError: Can't overwrite existing cassette ('/mnt/vss/_work/1/s/src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_uninstall_calico_npm.yaml') in your current record mode ('once').
> E           No match for the request (<Request (GET) [https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.ContainerService/locations/eastus/kubernetesVersions?api-version=2024-**05-01**>](https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.ContainerService/locations/eastus/kubernetesVersions?api-version=2024-05-01%3E)) was found.
> E           Found 1 similar requests with 1 different matcher(s) :
> E           
> E           1 - (<Request (GET) [https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.ContainerService/locations/eastus/kubernetesVersions?api-version=2024-**04-02-preview**>).](https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.ContainerService/locations/eastus/kubernetesVersions?api-version=2024-04-02-preview%3E).)

You could use the recording files from this pipeline artifact
https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=99088325&view=results
image

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**